qemu/tests/boot_with_different_vectors.py

```diff
@@ -45,41 +49,168 @@ def boot_guest_with_vectors(vectors):
+                queues_number = value_dict["queues"]
+                return (MSIs_number, queues_number)
+            except Exception:
+                pass
+        return (None, None)
+
+    def get_MSIs_and_queues_windows():
+        session_serial = vm.wait_for_serial_login(login_timeout)
+        # start traceview
+        error_context.context("Start trace view with pdb file", logging.info)
+        log_path = "c:\\logfile.etl"
+        clean_cmd = "del %s" % log_path
+        session_serial.cmd(clean_cmd)
+        traceview_local_path = "c:\\traceview.exe"
+        pdb_local_path = "c:\\netkvm.pdb"
+        start_traceview_cmd = "%s -start test_session -pdb %s -level 5 -flag 0x1fff -f %s"
+        start_traceview_cmd %= (traceview_local_path, pdb_local_path, log_path)
+        session_serial.cmd(start_traceview_cmd)
+
+        # restart nic
+        error_context.context("Restart guest nic", logging.info)
+        mac = vm.get_mac_address(0)
+        connection_id = utils_net.get_windows_nic_attribute(session_serial,
+                                                            "macaddress",
+                                                            mac,
+                                                            "netconnectionid")
+        utils_net.restart_windows_guest_network(session_serial, connection_id)
+
+        # stop traceview
+        error_context.context("Stop traceview", logging.info)
+        stop_traceview_cmd = "%s -stop test_session" % traceview_local_path
+        session_serial.cmd(stop_traceview_cmd)
+
+        # checkout traceview output
+        error_context.context("Check etl file generated by traceview", logging.info)
+        dump_file = "c:\\trace.txt"
+        clean_cmd = "del %s" % dump_file
+        session_serial.cmd(clean_cmd)
+        dump_cmd = "%s -process %s -pdb %s -o %s"
+        dump_cmd %= (traceview_local_path, log_path, pdb_local_path, dump_file)
+        status, output = session_serial.cmd_status_output(dump_cmd)
+        if status:
+            test.error("Cann't dump log file %s: %s" % (log_path, output))
+        time.sleep(60)
+        status, output = session_serial.cmd_status_output("taskkill /im traceview.exe && type %s" % dump_file)
+        if status:
+            test.error("Cann't read dumped file %s: %s" % (dump_file, output))
+
+        return get_MSI_queue_from_traceview_output(output)
+
+    def prepare_traceview_windows(session, timeout):
+        # verify driver
+        error_context.context("Check if the driver is installed and "
+                              "verified", logging.info)
+        driver_name = params.get("driver_name", "netkvm")
+        session = utils_test.qemu.windrv_check_running_verifier(session, vm,
+                                                                test,
+                                                                driver_name,
+                                                                timeout)
+
+        # copy traceview.exe
```

Comment:
also can use the absolute path but not copy to c:\，right?

Reply:
Yes, but I think copy to c:\ is more stable.